tests/fault_quarantine_circuit_breaker_test.go

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"os"
+	"testing"
+	"tests/helpers"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestCircuitBreaker(t *testing.T) {
+	feature := features.New("TestCircuitBreaker").
+		WithLabel("suite", "fault-quarantine-circuit-breaker")
+
+	var testCtx *helpers.QuarantineTestContext
+	var originalCBState string
+	var originalDeployment *appsv1.Deployment
+	var testNodes []string
+	var threshold int
+
+	feature.Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		client, err := c.NewClient()
+		require.NoError(t, err)
+
+
+		originalCBState = helpers.GetCircuitBreakerState(ctx, t, c)
+
+		var newCtx context.Context
+		newCtx, testCtx, originalDeployment = helpers.SetupQuarantineTestWithOptions(ctx, t, c,
+			"data/basic-matching-configmap.yaml",
+			&helpers.QuarantineSetupOptions{
+				CircuitBreakerPercentage: 20,
+				CircuitBreakerDuration:   "10m",
+				CircuitBreakerState:      "CLOSED",
+			})
+
+		nodes, err := helpers.GetAllNodesNames(ctx, client)
+		require.NoError(t, err)
+
+		startIdx := int(float64(len(nodes)) * 0.50)
+		if startIdx >= len(nodes) {
+			startIdx = len(nodes) - 1
+		}
+		testNodes = nodes[startIdx:]
+		t.Logf("Using %d test nodes for CB tests", len(testNodes))
+
+		return newCtx
+	})
+
+	feature.Assess("manually TRIPPED CB blocks normal cordoning", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		client, err := c.NewClient()
+		require.NoError(t, err)
+
+		helpers.SetCircuitBreakerState(ctx, t, c, "TRIPPED")
+
+		event := helpers.NewHealthEvent(testCtx.NodeName).
+			WithErrorCode("79").
+			WithMessage("XID error occurred")
+		tempFile, err := helpers.SendHealthEventWithTemplate(testCtx.NodeName, event)
+		require.NoError(t, err)
+		defer os.Remove(tempFile)
+
+		t.Log("Verifying node NOT cordoned when CB is manually TRIPPED")
+
+		require.Never(t, func() bool {
```

Comment:
I like the helpers you created for quarantine, can we do the same? One for node never quarantined?